test/streaming-client.spec.ts

```diff
@@ -0,0 +1,176 @@
+        });
+
+        it('Returns duplex stream', () => {
+            // Act
+            const res = sut.start();
+
+            // Assert
+            expect(res).toBeInstanceOf(BufferedDuplex);
+        });
+
+        it('adds event listener to httpResponse', () => {
+            // Setup
+            const res = sut.start();
+            let statusCode = null;
+            sut.on('httpResponse', code => {
+                statusCode = code;
+            });
+
+            // Act
+            mockClient.emit('httpResponse', {statusCode: 401});
+
+            // Assert
+            expect(statusCode).toBe(401);
+        });
+
+        it('adds event listener to connectFailed', () => {
+            // Setup
+            const res = sut.start();
+            let connectionError = null;
+            let expectedError = new Error("fake error")
+            sut.on('connectFailed', error => {
+                connectionError = error;
+            });
+            
+            // Act
+            mockClient.emit('connectFailed', expectedError);
+
+            // Assert
+            expect(connectionError).toBe(expectedError);
+        });
+
+        it('adds event listener to connection close', () => {
+            // Setup
+            const res = sut.start();
+            let closeCode = null;
+            let closeReason = null;
+            const expectedCloseCode = 1000;
+            const expectedCloseReason = "NormalClosure";
+            let mockConnection = new WebSocketConnection();
+            sut.on('close', (code, reason) => {
+                closeCode = code;
+                closeReason = reason;
+            });
+
+            // Act
+            mockClient.emit('connect', mockConnection);
+            mockConnection.emit('close', expectedCloseCode, expectedCloseReason);
+
+            // Assert
+            expect(closeCode).toBe(expectedCloseCode);
+            expect(closeReason).toBe(expectedCloseReason);
+        });
+
+        it('adds event listener to connection error', () => {
+            // Setup
+            const res = sut.start();
+            let connectionError = null;
+            const expectedError = new Error("fake connection error");
+            let mockConnection = new WebSocketConnection();
+            sut.on('error', error => {
+                connectionError = error;
+            });
+
+            // Act
+            mockClient.emit('connect', mockConnection);
+            mockConnection.emit('error', expectedError);
+
+            // Assert
+            expect(connectionError).toBe(expectedError);
+        });
+
+        it('emits connected event on connected message from server', () => {
+            // Setup
+            const res = sut.start();
+            let jobId = null;
+            const expectedJobId = "1";
+            let mockConnection = new WebSocketConnection();
+            sut.on('connect', response => {
+                jobId = response.id;
```

Comment:
do we get any other information in response?

We might as well validate all of the properties on response, not just id

Reply:
I see that you also pass `type`

Reply:
We get no other information. The connected response only has a job id right now

Reply:
In this test you are specifying the properties that are returned: you pass type and jobId